fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/api/DatatablesApiResource.java

```diff
@@ -218,6 +226,55 @@ public String getDatatable(@PathParam("datatable") @Parameter(description = "dat
         return this.toApiJsonSerializer.serializePretty(prettyPrint, result);
     }
 
+    @GET
+    @Path("{datatable}/query")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Query Data Table values", description = "Query values from a registered data table.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = List.class))) })
+    public String queryValues(@PathParam("datatable") @Parameter(description = "datatable") final String datatable,
+            @QueryParam("columnFilter") @Parameter(description = "columnFilter") final String columnFilter,
+            @QueryParam("valueFilter") @Parameter(description = "valueFilter") final String valueFilter,
+            @QueryParam("resultColumns") @Parameter(description = "resultColumns") final String resultColumns,
+            @Context final UriInfo uriInfo) {
+        this.context.authenticatedUser().validateHasDatatableReadPermission(datatable);
+        List<String> dataTableColumnNames = genericDataService.fillResultsetColumnHeaders(datatable).stream()
```

Comment:
I think we should move this logic one layer deeper and not have it on the API level, at least according to the current practices in fineract.

Reply:
I don't agree why are we validating input params on database service layer instead at API level where it should occour but if the project is doing that way, I am moving it there.

Reply:
It's a matter of design overall in the codebase. It's one level deeper and as I said in another comment, right now the validation logic is scattered between this level and the service layer.